### PR TITLE
Setup container environment variables before user

### LINKED ISSUE #3186: cri-o must set home ton $HOME if specified in image
When cri-o is used  in OpenShift, it automatically creates the arbitrary user that OpenShift requires to run a specific image. as an example:
```
sh-4.2$ tail -1 /etc/passwd
1000580000:x:1000580000:0:1000580000 user:/:/sbin/nologin
sh-4.2$
```
However, the home directory is set to "/"  .
```
sh-4.2$ getent passwd $(id -u) | cut -f6 -d:
/
```

When the $HOME environment variable is set in the built container image, it is preferable that the user home matches the same value.

### PR DESCRIPTION
**Pull request by mrunalp**
The user setup code looks at the env to setup HOME.

Fixes #3186 

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by haircommander** [comment]
LGTM

File: server/container_create_linux.go [hunk 1]
```diff
@@ -758,6 +758,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID, contai
 		specgen.AddAnnotation("org.opencontainers.image.stopSignal", containerImageConfig.Config.StopSignal)
 	}
 
+	// Add environment variables from image the CRI configuration
+	envs := mergeEnvs(containerImageConfig, containerConfig.GetEnvs())
+	for _, e := range envs {
+		parts := strings.SplitN(e, "=", 2)
+		specgen.AddProcessEnv(parts[0], parts[1])
```
Comment (TomSweeneyRedHat) [inline server/container_create_linux.go]: as noted elsewhere, possible. albeit unlikely, panic here if there's not an equal sign.

**Pull request merged**